test/jldtests.jl

```diff
@@ -448,6 +507,10 @@ for compress in (true,false)
         @check fidr d
         exr = read(fidr, "ex")   # line numbers are stripped, don't expect equality
         checkexpr(ex, exr)
+        funr = read(fidr, "fun")
+        checkfuns(fun, funr)
+        function_referencing_module_r = read(fidr, "function_referencing_module")
+        checkfuns(function_referencing_module, function_referencing_module_r)
```

Comment:
All of this is really nicely done.

Unless I'm missing something, you don't test that you can call a function that you load from a file. While it's trivial (you've checked the ASTs very carefully), it seems like a good sanity check.